services/mesh-discovery/pkg/discovery/mesh-workload/k8s/appmesh/appmesh_mesh_workload_scanner.go

```diff
@@ -124,3 +130,18 @@ func (a *appMeshWorkloadScanner) buildMeshWorkloadName(deploymentName string, na
 	// TODO: https://github.com/solo-io/service-mesh-hub/issues/141
 	return fmt.Sprintf("%s-%s-%s-%s", "appmesh", deploymentName, namespace, clusterName)
 }
+
+func (a *appMeshWorkloadScanner) getContainerPorts(
+	deployment *k8s_apps_types.Deployment,
+) []*zephyr_discovery_types.MeshWorkloadSpec_Appmesh_ContainerPort {
+	var ports []*zephyr_discovery_types.MeshWorkloadSpec_Appmesh_ContainerPort
```

Comment:
super nit: declare capacity since we have it.
```suggestion
	ports := make([]*zephyr_discovery_types.MeshWorkloadSpec_Appmesh_ContainerPort, 0, len(container.Ports))
```

Reply:
the length is actually the number of container ports across all containers, computing that would require looping through the containers anyways